tests/test_fsck.sh

```diff
@@ -1,75 +1,74 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 TESTCASE_DIR=$1
 IMAGE_FILE=exfat.img
 FSCK_PROG=../build/sbin/fsck.exfat
 FSCK_OPTS=-y
 PASS_COUNT=0
 
-function cleanup {
+cleanup() {
 	echo ""
 	echo "Passed ${PASS_COUNT} of ${TEST_COUNT}"
 	exit
 }
 
 if [ $# -eq 0 ]; then
-	TESTCASE_DIRS=`find -mindepth 1 -maxdepth 1 -type d`
-	TEST_COUNT=`find -mindepth 1 -maxdepth 1 -type d | wc -l`
+	TESTCASE_DIRS=$(find . -mindepth 1 -maxdepth 1 -type d)
+	TEST_COUNT=$(find . -mindepth 1 -maxdepth 1 -type d | wc -l)
 else
 	TESTCASE_DIRS=$@
 	TEST_COUNT=$#
 fi
 
 for TESTCASE_DIR in $TESTCASE_DIRS
 do
-	if [ ! -e ${TESTCASE_DIR}/${IMAGE_FILE}.tar.xz ]; then
+	if [ ! -e "${TESTCASE_DIR}/${IMAGE_FILE}.tar.xz" ]; then
 		TEST_COUNT=$((TEST_COUNT - 1))
 		continue
 	fi
 
-	echo "Running $TESTCASE_DIR"
+	echo "Running ${TESTCASE_DIR}"
 	echo "-----------------------------------"
 
 	# Set up image file as loop device
-	tar -C . -xf $TESTCASE_DIR/$IMAGE_FILE.tar.xz
-	losetup -f $IMAGE_FILE
-	DEV_FILE=`losetup -j $IMAGE_FILE | awk '{print $1}' | sed 's/://g'`
+	tar -C . -xf "${TESTCASE_DIR}/${IMAGE_FILE}.tar.xz"
+	DEV_FILE=$(losetup -f "${IMAGE_FILE}" --show)
 
 	# Run fsck for repair
 	$FSCK_PROG $FSCK_OPTS $DEV_FILE
-	if [ "$?" -ne "1" ]; then
+	if [ "$?" -ne 1 ]; then
```

Comment:
is the double quote around $? needed?

Reply:
It is not, $? is guaranteed to always return an integer, and shellcheck won't warn about it being unquoted.

I could have cleaned up the scripts file even more, but decided to do the least amount of changes for now.